### DISCUSSION TITLE
Image Polygonal Annotation (polygon, rectangle)

### DISCUSSION THREAD

**Discussion by monkeycc** [GraphiteEditor/Graphite#1354] [General]
![annotation](https://github.com/GraphiteEditor/Graphite/assets/6490927/03638b90-f6b0-4781-a0a6-bbb8b5fd97d9)
https://github.com/wkentaro/labelme

1. Batch import of images
2. Add labels for each box
3. Export coordinate file

I feel like it can be achieved

But their front-end and rust level cannot meet the requirements

Hope someone is interested and can achieve it

**Comment by Keavon** [comment]
Could you expand on the motivation and use case? Is this for image segmentation human labeling?

**Comment by monkeycc** [comment]
image segmentation labeling
yes

Polygon annotation
Rectangle annotation

image annotation
Audio annotation
Text annotation


Now the image annotation tool

https://github.com/HumanSignal/label-studio

## Link
| Tool           | Link                                 |
| -------------------- | ------------------------------------------- |
| RectTool     | https://labelbee.github.io                  |
| PolygonTool | https://labelbee.github.io?tool=polygonTool |
| LineTool      | https://labelbee.github.io?tool=lineTool    |
| PointTool   | https://labelbee.github.io?tool=pointTool   |
| TextTool      | https://labelbee.github.io?tool=textTool    |
| TagTaool       | https://labelbee.github.io?tool=tagTool     |


Not easy to use

1. The frame is not smooth
2. Keyboard and mouse control is not easy to use



Graphite can solve performance issues
Smooth Graphite Frame
Graphite controls keyboard and mouse smoothly


Artificial intelligence is very popular now

But there are issues with annotation tools

I think Graphite secondary development

Change to Image Annotation Tool

Can effectively solve these problems



Artificial intelligence annotation tools compared to image editing tools
Artificial intelligence will be more promising
The demand will be even greater

**Reply by Keavon** [reply]
I don't think that use case makes sense fitting within the scope of Graphite's vision. Human labeling tools require a highly efficient, ultra fast workflow to let the humans perform the repetitive task with utmost speed and efficiency. Graphite won't ever be able to serve that purpose since it's not tailor-made for that individual use case. I think it's safe to say this is explicitly a non-goal for this project.
